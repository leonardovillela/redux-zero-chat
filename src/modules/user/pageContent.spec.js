import React from 'react';
import { shallow } from 'enzyme';
import Chance from 'chance';

import CreateUser from './pageContent';

describe('CreateUserPage unit', () => {
  let chance = new Chance();

  it('Should render with success', () => {
    const wrapperWithUserName = shallow(<CreateUser
      userName="User Test"
      setUsername={() => {}}
      createUser={() => {}}
    />);

    const wrapperWithoutUserName = shallow(<CreateUser
      setUsername={() => {}}
      createUser={() => {}}
    />);

    expect(wrapperWithUserName).toMatchSnapshot();
    expect(wrapperWithoutUserName).toMatchSnapshot();
  });

  it('Should call onChange correctly', () => {
    const setUsernameMock = jest.fn();

    const nameStub = chance.name();
    const event = { target: { name: 'change', value: nameStub } };

    const wrapper = shallow(<CreateUser
      setUsername={setUsernameMock}
    />);

    wrapper.find('input').simulate('change', event);
    expect(setUsernameMock).toBeCalledWith(nameStub);
  })

  it('Should call onSubmit correctly', () => {
    const createUserMock = jest.fn();
    const preventDefaultMock = jest.fn();

    const event = { target: { name: 'submit' }, preventDefault: preventDefaultMock };

    const wrapper = shallow(<CreateUser createUser={createUserMock}/>);

    wrapper.find('form').simulate('submit', event);

    expect(preventDefaultMock).toBeCalled();
    expect(createUserMock).toBeCalled();
  });

  it('Should render submitButton in disabled state when username is empty', () => {
    const wrapper = shallow(<CreateUser userName="" />);
    expect(wrapper.find("button[type='submit']").prop('disabled')).toBeTruthy();
  });

  it('Should render submitButton in enabled state when username is not empty', () => {
    const wrapper = shallow(<CreateUser userName={chance.name()} />);
    expect(wrapper.find("button[type='submit']").prop('disabled')).toBeFalsy();
  });
});
