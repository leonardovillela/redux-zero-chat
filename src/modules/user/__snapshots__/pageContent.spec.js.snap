// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUserPage unit Should render with success 1`] = `
<div
  className="w-60 center"
>
  <h1
    className="tc"
  >
    Digite seu nome
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      className="w-100 pa2"
      onChange={[Function]}
      type="text"
      value="User Test"
    />
    <button
      className="w-100 pa3 mt3"
      disabled={false}
      type="submit"
    >
      Entrar
    </button>
  </form>
</div>
`;

exports[`CreateUserPage unit Should render with success 2`] = `
<div
  className="w-60 center"
>
  <h1
    className="tc"
  >
    Digite seu nome
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      className="w-100 pa2"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="w-100 pa3 mt3"
      disabled={true}
      type="submit"
    >
      Entrar
    </button>
  </form>
</div>
`;
